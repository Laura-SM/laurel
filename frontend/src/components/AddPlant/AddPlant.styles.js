import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  centralContainer: {
    alignItems: 'center',
  },
  image: {
    width: 230,
    height: 230,
    borderRadius: 10,
    margin: 20,
    backgroundColor: '#E5E5E5',
  },
  featuresContainer: {
    width: 320,
    alignItems: 'flex-start',
  },
  iconsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 10,
  },
  input: {
    width: 250,
    height: 45,
    backgroundColor: '#E5E5E5',
    borderRadius: 10,
    fontSize: 20,
    paddingHorizontal: 20,
    textAlignVertical: 'center',
    marginHorizontal: 10,
  },
});

export default styles;
