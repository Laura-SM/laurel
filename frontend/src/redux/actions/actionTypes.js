export default {
  LOAD_PLANTS: 'LOAD_PLANTS',
  ADD_PLANT: 'ADD_PLANT',
  UPDATE_PLANT: 'UPDATE_PLANT',
  DELETE_PLANT: 'DELETE_PLANT',
  LOAD_PLANT: 'LOAD_PLANT',
  PLANTS_ERROR: 'PLANTS_ERROR',

  GET_USER: 'GET_USER',
  USER_ERROR: 'USER_ERROR',
};
