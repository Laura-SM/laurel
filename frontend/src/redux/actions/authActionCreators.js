import axios from 'axios';
import actionTypes from './actionTypes';
import {env} from '../../../.env.js';

const url = env.REACT_APP_PORT;

export function signInUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/signin`, user);
      dispatch({
        type: actionTypes.SIGNIN_USER,
        userTokens: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}

export function signUpUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/signup`, user);
      dispatch({
        type: actionTypes.SIGNUP_USER,
        user: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}
