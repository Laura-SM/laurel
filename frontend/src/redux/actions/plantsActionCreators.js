import axios from 'axios';
import actionTypes from './actionTypes';
import {env} from '../../../.env.js';

const url = env.REACT_APP_PORT;

export function loadPlants() {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/plants`);
      dispatch({
        type: actionTypes.LOAD_PLANTS,
        plants: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function addPlant(plant) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/plants`, plant);
      dispatch({
        type: actionTypes.ADD_PLANT,
        plant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function deletePlant(plantId) {
  return async dispatch => {
    try {
      await axios.delete(`${url}/plants/${plantId}`);
      dispatch({
        type: actionTypes.DELETE_PLANT,
        plantId,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function updatePlant(plant) {
  return async dispatch => {
    try {
      const {data} = await axios.put(`${url}/plants/${plant._id}`, plant);
      dispatch({
        type: actionTypes.UPDATE_PLANT,
        plant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function loadPlant(plantId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/plants/${plantId}`);
      dispatch({
        type: actionTypes.LOAD_PLANT,
        selectedPlant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}
