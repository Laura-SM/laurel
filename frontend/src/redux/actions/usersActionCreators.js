import axios from 'axios';
import actionTypes from './actionTypes';

const url = 'http://192.168.8.104:2021/login';

export function loadUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(url, user);
      dispatch({
        type: actionTypes.SIGNIN_USER,
        userTokens: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}
