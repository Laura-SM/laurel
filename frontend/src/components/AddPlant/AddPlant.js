import React, {useState} from 'react';
import {Text, TextInput, View, TouchableOpacity} from 'react-native';
import {connect} from 'react-redux';
import {addPlant} from '../../redux/actions/plantsActionCreators';
import {updateUser} from '../../redux/actions/usersActionCreators';
import {useNavigation} from '@react-navigation/native';
import globalStyles from '../../styles/global.styles';
import styles from './AddPlant.styles';

const AddPlant = ({dispatch, userAccess, selectedPlant}) => {
  const navigation = useNavigation();
  let [name, setPlantNameInputValue] = useState('');
  let [room, setRoomInputValue] = useState('');
  let [nextWaterDate, setNextWaterDateInputValue] = useState('');
  let [nextMistDate, setNextMistDateInputValue] = useState('');
  let [nextTransplantDate, setNextTransplantDateInputValue] = useState('');
  let newPlant = {
    name,
    room,
    nextWaterDate,
    nextMistDate,
    nextTransplantDate,
    card: false,
  };

  const onPressAddPlant = () => {
    dispatch(addPlant(newPlant));
  };

  const userPlantsIds = userAccess.user.plants;
  const updateUserPlantsAndGoBack = () => {
    userPlantsIds.push(selectedPlant._id);
    dispatch(updateUser(userAccess.user));
    navigation.navigate('MyPlants', {myPlantsIds: userPlantsIds});
  };

  return (
    <View>
      <Text>Add plant</Text>
      <TouchableOpacity
        style={globalStyles.roundButton}
        onPress={onPressAddPlant}>
        <Text style={styles.textButton}>Add</Text>
      </TouchableOpacity>
      <TextInput
        style={styles.input}
        onChangeText={text => setPlantNameInputValue(text)}
        value={name}
        placeholder="Plant name"
        keyboardType="default"
      />
      <TextInput
        style={styles.input}
        onChangeText={text => setRoomInputValue(text)}
        value={room}
        placeholder="Room"
        keyboardType="default"
      />
      <TextInput
        style={styles.input}
        onChangeText={text => setNextWaterDateInputValue(text)}
        value={nextWaterDate}
        placeholder="Last watering date"
        keyboardType="default"
      />
      <TextInput
        style={styles.input}
        onChangeText={text => setNextMistDateInputValue(text)}
        value={nextMistDate}
        placeholder="Last misting date"
        keyboardType="default"
      />
      <TextInput
        style={styles.input}
        onChangeText={text => setNextTransplantDateInputValue(text)}
        value={nextTransplantDate}
        placeholder="Last transplant date"
        keyboardType="default"
      />
      <TouchableOpacity
        style={globalStyles.roundButton}
        onPress={() => updateUserPlantsAndGoBack()}>
        <Text>Upd</Text>
      </TouchableOpacity>
    </View>
  );
};

function mapStateToProps({userAccess, selectedPlant}) {
  return {
    userAccess,
    selectedPlant,
  };
}

export default connect(mapStateToProps)(AddPlant);
