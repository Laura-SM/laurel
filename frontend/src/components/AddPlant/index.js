import React, {useState} from 'react';
import {
  Text,
  TextInput,
  View,
  TouchableOpacity,
  Image,
  ScrollView,
  Pressable,
} from 'react-native';
import {connect} from 'react-redux';
import {addPlant} from '../../redux/actions/plantsActionCreators';
import {updateUser} from '../../redux/actions/usersActionCreators';
import DateTimePicker from '@react-native-community/datetimepicker';
import globalStyles from '../../styles/globalStyles';
import styles from './styles';

const AddPlant = ({dispatch, userAccess, selectedPlant, navigation}) => {
  const [name, setPlantNameInputValue] = useState('');
  const [room, setRoomInputValue] = useState('');

  const [nextWaterDate, setNextWaterDateInputValue] = useState(new Date());
  const [showWaterDatePicker, setShowWaterDatePicker] = useState(false);

  const [nextMistDate, setNextMistDateInputValue] = useState(new Date());
  const [showMistDatePicker, setShowMistDatePicker] = useState(false);

  const [nextTransplantDate, setNextTransplantDateInputValue] = useState(
    new Date(),
  );
  const [showTransplantDatePicker, setShowTransplantDatePicker] =
    useState(false);

  const onPressAddPlant = () => {
    let newPlant = {
      name,
      room,
      nextWaterDate: new Date(
        Date.UTC(
          nextWaterDate.getFullYear(),
          nextWaterDate.getMonth(),
          nextWaterDate.getDate(),
        ),
      ),
      nextMistDate: new Date(
        Date.UTC(
          nextMistDate.getFullYear(),
          nextMistDate.getMonth(),
          nextMistDate.getDate(),
        ),
      ),
      nextTransplantDate: new Date(
        Date.UTC(
          nextTransplantDate.getFullYear(),
          nextTransplantDate.getMonth(),
          nextTransplantDate.getDate(),
        ),
      ),
      card: false,
    };

    dispatch(addPlant(newPlant));
  };
  const userPlantsIds = userAccess.user.plants;
  const updateUserPlantsAndGoBack = () => {
    if (selectedPlant.card) {
      navigation.navigate('MyPlants', {myPlantsIds: userPlantsIds});
    } else {
      if (userPlantsIds.includes(selectedPlant._id)) {
        navigation.navigate('MyPlants', {myPlantsIds: userPlantsIds});
      } else {
        userPlantsIds.push(selectedPlant._id);
        dispatch(updateUser(userAccess.user));
        navigation.navigate('MyPlants', {myPlantsIds: userPlantsIds});
      }
    }
  };

  return (
    <ScrollView style={globalStyles.mainContainer}>
      <View style={globalStyles.headerContainer}>
        <Text style={globalStyles.titleText}>Add plant</Text>
      </View>
      <View style={styles.centralContainer}>
        <Text style={styles.imageText}>Add a photo of your plant</Text>
        <Image
          style={styles.image}
          source={require('../../icons/flower48.png')}
        />
        <View style={styles.featuresContainer}>
          <View style={styles.iconsContainer}>
            <Image source={require('../../icons/description24.png')} />
            <TextInput
              style={styles.input}
              onChangeText={text => setPlantNameInputValue(text)}
              value={name}
              placeholder="Plant name"
              keyboardType="default"
            />
          </View>
          <View style={styles.iconsContainer}>
            <Image source={require('../../icons/room24.png')} />
            <TextInput
              style={styles.input}
              onChangeText={text => setRoomInputValue(text)}
              value={room}
              placeholder="Room"
              keyboardType="default"
            />
          </View>
          <View style={styles.iconsContainer}>
            <Image source={require('../../icons/water24.png')} />
            <Pressable onPress={() => setShowWaterDatePicker(true)}>
              <TextInput
                style={styles.input}
                value={nextWaterDate.toLocaleDateString()}
                editable={false}
              />
            </Pressable>
            <>
              {showWaterDatePicker && (
                <DateTimePicker
                  testID="dateTimePicker"
                  value={nextWaterDate}
                  mode="date"
                  is24Hour={true}
                  display="default"
                  onChange={(e, selectedDate) => {
                    setShowWaterDatePicker(false);
                    const currentDate = selectedDate || nextWaterDate;
                    setNextWaterDateInputValue(currentDate);
                  }}
                />
              )}
            </>
          </View>
          <View style={styles.iconsContainer}>
            <Image source={require('../../icons/mist24.png')} />
            <Pressable onPress={() => setShowMistDatePicker(true)}>
              <TextInput
                style={styles.input}
                value={nextMistDate.toLocaleDateString()}
                editable={false}
              />
            </Pressable>
            <>
              {showMistDatePicker && (
                <DateTimePicker
                  testID="dateTimePicker"
                  value={nextMistDate}
                  mode="date"
                  is24Hour={true}
                  display="default"
                  onChange={(e, selectedDate) => {
                    setShowMistDatePicker(false);
                    const currentDate = selectedDate || nextMistDate;
                    setNextMistDateInputValue(currentDate);
                  }}
                />
              )}
            </>
          </View>
          <View style={styles.iconsContainer}>
            <Image source={require('../../icons/settings24.png')} />
            <Pressable onPress={() => setShowTransplantDatePicker(true)}>
              <TextInput
                style={styles.input}
                value={nextTransplantDate.toLocaleDateString()}
                editable={false}
              />
            </Pressable>
            <>
              {showTransplantDatePicker && (
                <DateTimePicker
                  testID="dateTimePicker"
                  value={nextTransplantDate}
                  mode="date"
                  is24Hour={true}
                  display="default"
                  onChange={(e, selectedDate) => {
                    setShowTransplantDatePicker(false);
                    const currentDate = selectedDate || nextTransplantDate;
                    setNextTransplantDateInputValue(currentDate);
                  }}
                />
              )}
            </>
          </View>
        </View>
        <TouchableOpacity
          testID="addPlantButton"
          style={globalStyles.submitButton}
          onPress={onPressAddPlant}>
          <Image source={require('../../icons/done24.png')} />
        </TouchableOpacity>
      </View>
      <View style={globalStyles.bottomContainer}>
        <TouchableOpacity
          testID="goBackButton"
          style={globalStyles.roundButton}
          onPress={() => updateUserPlantsAndGoBack()}>
          <Image source={require('../../icons/goBack24.png')} />
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
};

function mapStateToProps({userAccess, selectedPlant}) {
  return {
    userAccess,
    selectedPlant,
  };
}

export default connect(mapStateToProps)(AddPlant);
