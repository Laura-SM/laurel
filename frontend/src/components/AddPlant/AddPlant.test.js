import React from 'react';
import AddPlant from './index';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import * as plantsActions from '../../redux/actions/plantsActionCreators';
import * as usersActions from '../../redux/actions/usersActionCreators';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a AddPlant component', () => {
  const mockStore = configureStore([thunk]);
  let store = mockStore({
    selectedPlant: {name: 'kentia', _id: '2b', card: false},
    userAccess: {
      user: {
        plants: ['1a'],
      },
    },
  });
  const navigation = {
    navigate: jest.fn(),
  };

  describe('When is AddPlant function is invoked', () => {
    test('Then AddPlant is rendered correctly', () => {
      const wrapper = render(
        <Provider store={store}>
          <AddPlant />
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When done button is pressed', () => {
    test('Then dispatch addPlant', () => {
      jest.spyOn(plantsActions, 'addPlant').mockReturnValueOnce({type: ''});
      const {getByTestId} = render(
        <Provider store={store}>
          <AddPlant />
        </Provider>,
      );
      const addPlant = getByTestId('addPlantButton');
      fireEvent.press(addPlant);

      expect(plantsActions.addPlant).toHaveBeenCalled();
    });
  });

  describe('When goBack button is pressed with a new plant', () => {
    test('Then dispatch updateUser', () => {
      jest.spyOn(usersActions, 'updateUser').mockReturnValueOnce({type: ''});
      const {getByTestId} = render(
        <Provider store={store}>
          <AddPlant navigation={navigation} />
        </Provider>,
      );
      const goBack = getByTestId('goBackButton');
      fireEvent.press(goBack);

      expect(usersActions.updateUser).toHaveBeenCalled();
    });
  });

  describe('When goBack button is pressed with a repeat plant', () => {
    test('Then dispatch updateUser', () => {
      jest.spyOn(usersActions, 'updateUser').mockReturnValueOnce({type: ''});
      store = mockStore({
        selectedPlant: {name: 'kentia', _id: '1a', card: false},
        userAccess: {
          user: {
            plants: ['1a'],
          },
        },
      });
      const {getByTestId} = render(
        <Provider store={store}>
          <AddPlant navigation={navigation} />
        </Provider>,
      );
      const goBack = getByTestId('goBackButton');
      fireEvent.press(goBack);

      expect(usersActions.updateUser).toHaveBeenCalled();
    });
  });

  describe('When goBack button is pressed with a card plant', () => {
    test('Then navigation.navigate is invoked', () => {
      store = mockStore({
        selectedPlant: {name: 'kentia', _id: '1a', card: true},
        userAccess: {
          user: {
            plants: ['2a'],
          },
        },
      });
      const {getByTestId} = render(
        <Provider store={store}>
          <AddPlant navigation={navigation} />
        </Provider>,
      );
      const goBack = getByTestId('goBackButton');
      fireEvent.press(goBack);

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
