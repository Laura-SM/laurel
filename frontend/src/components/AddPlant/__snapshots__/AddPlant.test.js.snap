// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a AddPlant component When is AddPlant function is invoked Then AddPlant is rendered correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 20,
          }
        }
      >
        Add plant
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "left": 95,
            "position": "absolute",
            "top": 125,
            "zIndex": 1,
          }
        }
      >
        Add a photo of your plant
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/icons/flower48.png",
          }
        }
        style={
          Object {
            "backgroundColor": "#E5E5E5",
            "borderRadius": 10,
            "height": 230,
            "margin": 20,
            "width": 230,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "width": 320,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/icons/description24.png",
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Plant name"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#E5E5E5",
                "borderRadius": 10,
                "fontSize": 20,
                "height": 45,
                "marginHorizontal": 10,
                "paddingHorizontal": 20,
                "textAlignVertical": "center",
                "width": 250,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/icons/room24.png",
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Room"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#E5E5E5",
                "borderRadius": 10,
                "fontSize": 20,
                "height": 45,
                "marginHorizontal": 10,
                "paddingHorizontal": 20,
                "textAlignVertical": "center",
                "width": 250,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/icons/water24.png",
              }
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <TextInput
              allowFontScaling={true}
              editable={false}
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#E5E5E5",
                  "borderRadius": 10,
                  "fontSize": 20,
                  "height": 45,
                  "marginHorizontal": 10,
                  "paddingHorizontal": 20,
                  "textAlignVertical": "center",
                  "width": 250,
                }
              }
              underlineColorAndroid="transparent"
              value="16/06/2021"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/icons/mist24.png",
              }
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <TextInput
              allowFontScaling={true}
              editable={false}
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#E5E5E5",
                  "borderRadius": 10,
                  "fontSize": 20,
                  "height": 45,
                  "marginHorizontal": 10,
                  "paddingHorizontal": 20,
                  "textAlignVertical": "center",
                  "width": 250,
                }
              }
              underlineColorAndroid="transparent"
              value="16/06/2021"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/icons/settings24.png",
              }
            }
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <TextInput
              allowFontScaling={true}
              editable={false}
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#E5E5E5",
                  "borderRadius": 10,
                  "fontSize": 20,
                  "height": 45,
                  "marginHorizontal": 10,
                  "paddingHorizontal": 20,
                  "textAlignVertical": "center",
                  "width": 250,
                }
              }
              underlineColorAndroid="transparent"
              value="16/06/2021"
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#B8E0D2",
            "borderRadius": 10,
            "height": 65,
            "justifyContent": "center",
            "margin": 20,
            "opacity": 1,
            "width": 150,
          }
        }
        testID="addPlantButton"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/icons/done24.png",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#B8E0D2",
            "borderRadius": 50,
            "height": 60,
            "justifyContent": "center",
            "margin": 20,
            "opacity": 1,
            "width": 60,
          }
        }
        testID="goBackButton"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/icons/goBack24.png",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
